src/memory.cpp

```diff
@@ -119,7 +147,9 @@ inline namespace CUPLA_ACCELERATOR_NAMESPACE
     CUPLA_HEADER_ONLY_FUNC_SPEC
     cuplaError_t cuplaFreeHost(void* ptr)
     {
-        if(cupla::manager::Memory<cupla::AccHost, cupla::AlpakaDim<1u>>::get().free(ptr))
+        if(nullptr)
+            return cuplaSuccess;
+        else if(cupla::manager::Memory<cupla::AccHost, cupla::AlpakaDim<1u>>::get().free(ptr))
             return cuplaSuccess;
         else
             return cuplaErrorMemoryAllocation;
```

Comment:
Unrelated to this PR, but thematically belongs: this error code should not be memory allocation, but invalid value. I think if we do not want to have a system of codes fully mirroring CUDA it's fine, but then imo we should have an "other" code that covers all the not implemented ones.

I would agrue for a user getting that hypothetical unspecific "other" is much better than a misattributed memory allocation here.

Reply:
Will be handled in a different PR.